OpenSim/Common/DataTable.h

```diff
@@ -843,8 +843,9 @@ class DataTable_ : public AbstractDataTable {
         const size_t totalWidth{std::accumulate(columnWidths.cbegin(),
                                                 columnWidths.cend(),
                                                 static_cast<size_t>(0))};
-        result.reserve((totalWidth * table.size() * excessAllocation) +
-                       result.capacity());
+        result.reserve(result.capacity() +
+            static_cast<unsigned>(std::roundf(
+                totalWidth * table.size() * excessAllocation)));
```

Comment:
The existing implementation is implicitly casting a `float` to an `unsigned`; the compiler warns that information is being lost.